test/testcafe/spec/ChallengeAuthenticatorPassword_spec.js

```diff
@@ -235,3 +243,11 @@ test.requestHooks(mockChallengeAuthenticatorPassword)('should show custom factor
   await t.expect(challengePasswordPage.getFactorPageHelpLinksLabel()).eql('custom factor page link');
   await t.expect(challengePasswordPage.getFactorPageHelpLink()).eql('https://acme.com/what-is-okta-autheticators');
 });
+
+test.requestHooks(mockChallengeAuthenticatorPasswordXSS)('should properly escape the identifier and avoid XSS', async t => {
+  const challengePasswordPage = await setup(t);
+  await checkA11y(t);
+  await t.expect(challengePasswordPage.getIdentifier()).eql(identifierXSS);
+  await t.expect(challengePasswordPage.identifierHasContenteditable()).eql(false);
```

Comment:
Clever. I assume you turned it from red to green to confirm the vuln existed in the first place?

Reply:
Thank you. Yes, I made sure the test was failing first and then applied the fix to verify it was good.